### PR TITLE
chore: add structure of project

### PR DESCRIPTION
**Pull request by GabinDDL**
Add all the structure of the project :
- gitignore
- ci
- dune settings
- readme

@ThomArr

### CODE REVIEW

File: .github-ci.yml [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
```
Comment (ThomArr) [inline .github-ci.yml:239]: I think it passes on gitlab, but not on github, I had the same problem on another project in ocaml.

Comment (ThomArr) [reply .github-ci.yml:239]: If you want to, maybe you can look at that link: https://github.com/ThomArr/Chess/blob/main/.github/workflows/ocaml.yml

File: .gitignore [hunk 2]
```diff
@@ -27,3 +27,4 @@ setup.log
 
 # Local OPAM switch
 _opam/
```
Comment (ThomArr) [inline .gitignore:29]:
```suggestion
_opam/
_build/
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,94 @@
 # Math-Info-L3-Project
 L3 math-info project on recoloring, in algorithmic.
+
+## Members
+
+| Name                | Gaufre username |
```
Comment (ThomArr) [inline README.md:6]: Maybe put our pseudo github? (@ThomArr and @GabinDDL)

File: .github-ci.yml (hunk 1 above)
Comment (ThomArr) [reply .github-ci.yml:239]: I think at least you’re going to have to add: .github/workflows

**Comment by ThomArr** [comment]
Also, I don’t have the rights to secure merge requests. Perhaps add the fact that it must be reviewed by at least 1 person before it can be merged?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,94 @@
...
+| :------------------ | :-------------: |
+| Thomas ARROUS       |     @arrous     |
+| Gabin DUDILLIEU     |    @gdudilli    |
```
Comment (ThomArr) [inline README.md:9]:
```suggestion
| Name                | Github username |
| :------------------ | :-------------: |
| Thomas ARROUS       |     @ThomArr      |
| Gabin DUDILLIEU     |    @GabinDDL)    |
```

File: .gitignore (hunk 2 above)
Comment (GabinDDL) [reply .gitignore:29]: the build repertory is already in the gitignore

**Comment by GabinDDL** [comment]
@ThomArr it's resolved, thanks

**Review by ThomArr — APPROVED** [review]

**Pull request merged**